### PR TITLE
[Bugfix][ROCm][MLA] Pass q/kv dtypes to get_mla_metadata_v1 in FP8 decode

### PR DESCRIPTION
**Pull request by peizhang56**
## Purpose

Fix a silent accuracy regression for MLA models served with an **FP8 KV cache** on **gfx950 (MI355X)** using the AITER MLA backend. With `amd-aiter == 0.1.16.post2`, Kimi-K2.5 at TP2 (32 q-heads/rank) produces garbage — gsm8k drops from ~0.93 to ~0.

### Root cause

`AiterMLAMetadataBuilder` builds the persistent split/reduce metadata for decode via `get_mla_metadata_v1`, but did **not** forward `dtype_q` / `dtype_kv` (they default to `None`). For the gfx950 fp8/fp8 `nhead=32`, `qlen=1` internal "fold-to-16" decode path, omitting the dtypes makes `get_mla_metadata_v1` emit **wrong split/reduce metadata**, which silently corrupts the decode output.

Note that `get_mla_metadata_info_v1` (which sizes the metadata buffers, a few lines above) was already given `q_dtype` / `kv_dtype`. This change makes the two metadata APIs consistent by passing the same dtypes to `get_mla_metadata_v1`.

This is a different fix from #46952. #46952 works around the symptom by padding the q head count `32 -> 64` so the (broken) fold path is avoided entirely; this PR fixes the root cause so the metadata for the fold path is correct. The fix is 6 lines and reuses values already computed in `__init__`.

### Affected configurations

Triggered by **gfx950 + FP8 KV cache + per-rank head count that takes the nhead->16 fold path** (verified: 32, i.e. TP2). bf16 KV cache and the native head counts (16/64/128) are unaffected; this change always forwards the dtypes, which is a no-op for paths that already worked.

## Test Plan

End-to-end, Kimi-K2.5-MXFP4, MI355X (gfx950), FP8 KV cache, TP2 (32 heads/rank):

```bash
vllm serve /shareddata/Kimi-K2.5-MXFP4 -tp 2 \
    --max-model-len 32768 --max-num-batched-tokens 16384 \
    --gpu-memory-utilization 0.95 --no-enable-prefix-caching \
    --trust-remote-code --kv-cache-dtype fp8 --async-scheduling

lm_eval --model local-completions \
    --model_args model=/shareddata/Kimi-K2.5-MXFP4,base_url=http://localhost:8000/v1/completions,num_concurrent=64,max_retries=3,tokenized_requests=False,trust_remote_code=True \
    --tasks gsm8k --num_fewshot 3
```

A/B control isolating the fix: toggle only the `dtype_q` / `dtype_kv` kwargs on the `get_mla_metadata_v1` call, keeping everything else identical, and re-run the eval above.

## Test Result

End-to-end gsm8k (3-shot), Kimi-K2.5-MXFP4, gfx950, TP2, FP8 KV cache:

| Config | flexible-extract | strict-match |
|---|---|---|
| Before (dtypes omitted) | 0.0106 | — |
| After (this PR) | **0.9212** | **0.9181** |

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Review by tjtanaa — APPROVED** [review]

File: tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    ]
+    assert not mismatched, (
+        "AITER MLA persistent decode metadata does not match the fp8/bf16 "
+        f"golden for fields {mismatched}; the builder must forward "
+        "dtype_q/dtype_kv to get_mla_metadata_v1."
+    )
+
+
+if __name__ == "__main__":
+    pytest.main([__file__, "-v"])
```
Comment (AndreasKaratzas) [inline tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py]: nit: I don't think this is necessary

File: tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+import importlib.util
+import types
+from unittest.mock import patch
+
+import pytest
+import torch
+
+from vllm.platforms import current_platform
+
+aiter_available = importlib.util.find_spec("aiter") is not None
```
Comment (AndreasKaratzas) [inline tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py]: I think there is a built in function for that like aiter_found_and_supported or something

File: tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+
+def _on_gfx950() -> bool:
+    if not (current_platform.is_rocm() and aiter_available):
+        return False
+    try:
```
Comment (AndreasKaratzas) [inline tests/kernels/attention/test_rocm_aiter_mla_decode_metadata.py]: If you are exiting on non-rocm platforms, then this should never fail (aka no try except should be necessary here)

**Review by AndreasKaratzas — APPROVED** [review]
Test looks good

**Comment by peizhang56** [comment]
Async Engine, Inputs, Utils, Worker, Config (CPU) failed, but it should not be related to this change.

  - This is a CPU job running tool-parser / reasoning-parser / multimodal / renderer tests — no ROCm, no MLA, no attention, no aiter.
  - My PR only touches rocm_aiter_mla.py, the gfx950 test file, and test-amd.yaml. None of those are imported by or affect test_structural_tag_registry.py.
  - It's a structural-tag schema drift (any_order field added upstream, test not updated) — fixing it belongs in a separate tool-parser PR

**Comment by Rohan138** [comment]
FYI @tjtanaa @dllehr-amd @AndreasKaratzas looks like the qh16 path has other accuracy issues discussed in https://github.com/ROCm/aiter/pull/3304, which introduces a native qh32 path instead. Unfortunately, this PR did not make it into AITER 0.1.16.post2, even though it was previously present in 0.1.15.post3.

This was also breaking DSR1 MXFP4 TP4 serving btw, thanks @stacyroberts for the repro
